File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sum = 0;
    void sumOfLeftLeaves_01(TreeNode* node) {
        if(!node) return;
        
        if(node->left && !node->left->left && !node->left->right) sum+= node->left->val;
        
        sumOfLeftLeaves_01(node->left);
        sumOfLeftLeaves_01(node->right);
    }
    int sumOfLeftLeaves(TreeNode* root) {
        if(!root) return 0;
        sumOfLeftLeaves_01(root);
        return sum;
    }
};

File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int sum = 0;
    public void sumOfLeftLeaves_01(TreeNode node) {
        //Base Case return
        if(node == null) return;
        //left Node exists -> Node's left child's left and right are null
        if(node.left != null && node.left.left == null && node.left.right == null) sum += node.left.val;
        
        //Recursive Call to left and right Nodes.
        sumOfLeftLeaves_01(node.left);
        sumOfLeftLeaves_01(node.right);
    }
    public int sumOfLeftLeaves(TreeNode root) {
        sumOfLeftLeaves_01(root);
        return sum;
    }
}